File: problems/Single Number II/singleNumber.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        var seenOnce = 0;
        var seenTwice = 0;

        foreach (var item in nums) {
            seenOnce = ~seenTwice & (seenOnce ^ item);
            seenTwice = ~seenOnce & (seenTwice ^ item);
        }

        return seenOnce;
    }
}

File: problems/Single Number II/singleNumber.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> hashmap = new HashMap<>();

        for (int num : nums) {
            hashmap.put(num, hashmap.getOrDefault(num, 0) + 1);
        }

        for (int k : hashmap.keySet()) {
            if (hashmap.get(k) == 1) {
                return k;
            }
        }

        return -1;
    }
}
